superset/assets/javascripts/explorev2/stores/fields.js

```diff
@@ -552,42 +556,52 @@ export const fields = {
   series: {
     type: 'SelectField',
     label: 'Series',
-    choices: [],
     default: null,
     description: 'Defines the grouping of entities. ' +
                  'Each series is shown as a specific color on the chart and ' +
                  'has a legend toggle',
+    mapStateToProps: (state) => ({
+      choices: (state.datasource) ? state.datasource.gb_cols : [],
+    }),
   },
 
   entity: {
     type: 'SelectField',
     label: 'Entity',
-    choices: [],
     default: null,
     description: 'This define the element to be plotted on the chart',
+    mapStateToProps: (state) => ({
+      choices: state.datasource.gb_cols,
```

Comment:
looks like we should check for datasource existence here too
`(state.datasource) ? state.datasource.gb_cols : [],`